main.js

```diff
@@ -89,29 +89,64 @@ function createWindow () {
     mainWindow.flashFrame(false);
   });
 
-  // Load locale
-  const locale = 'en'; // FIXME
-  const localeData = JSON.parse(fs.readFileSync(path.join(__dirname, '_locales', locale, 'messages.json'), 'utf-8'))
+  function loadLocale() {
+    // possible locales: https://github.com/electron/electron/blob/master/docs/api/locales.md
+    const locale = app.getLocale();
+
+    if (/^en-/.test(locale)) {
+      return 'en';
+    }
+
+    return locale;
+  }
+
+  function loadLocaleMessages(locale) {
+    const onDiskLocale = locale.replace('-', '_');
+    const targetFile = path.join(__dirname, '_locales', onDiskLocale, 'messages.json');
+
+    return JSON.parse(fs.readFileSync(targetFile, 'utf-8'))
+  }
+
+  // Load locale - if we can't load messages for the current locale, we default to 'en'
+  var locale = loadLocale();
+  var messages;
+  try {
+    messages = loadLocaleMessages(locale);
+  }
+  catch (e) {
+    console.log('Problem loading messages for locale ', locale, e.stack);
+    locale = 'en';
+    messages = loadLocaleMessages(locale);
+  }
+
+  // Ingested in preload.js via a sendSync call
   ipc.on('locale-data', function(event, arg) {
-    event.returnValue = localeData;
+    event.returnValue = messages;
   });
 
-  // and load the index.html of the app.
-  mainWindow.loadURL(url.format({
-    pathname: path.join(__dirname, 'background.html'),
-    protocol: 'file:',
-    slashes: true,
-    query: {
-      locale: locale,
-      version: package_json.version,
-      buildExpiration: config.get('buildExpiration'),
-      serverUrl: config.get('serverUrl'),
-      environment: environment,
-      node_version: process.versions.node
-    }
-  }))
+  function prepareURL(pathSegments) {
+    return url.format({
+      pathname: path.join.apply(null, pathSegments),
+      protocol: 'file:',
+      slashes: true,
+      query: {
+        locale: locale,
+        version: package_json.version,
+        buildExpiration: config.get('buildExpiration'),
+        serverUrl: config.get('serverUrl'),
+        environment: environment,
+        node_version: process.versions.node
+      }
+    })
+  }
+
+  if (process.env.NODE_ENV === 'test') {
+    mainWindow.loadURL(prepareURL([__dirname, 'test', 'index.html']));
+  } else {
+    mainWindow.loadURL(prepareURL([__dirname, 'background.html']));
+  }
 
-  if (config.get('openDevTools')) {
+  if (config.get('openDevTools') && !process.env.HIDE_DEV_TOOLS) {
```

Comment:
It would be cleaner to do this, but there's one problem. With a `local-test.json` file in place turning on the dev tools, `grunt unit-tests` will fully break. We'll time out when waiting for the test results, because we can't extract them via the `execute()` method. :0(